primitives/trie/src/storage_proof.rs

```diff
@@ -121,13 +120,10 @@ impl CompactProof {
 	}
 
 	/// Decode to a full storage_proof.
-	///
-	/// Method use a temporary `HashDB`, and `sp_trie::decode_compact`
-	/// is often better.
```

Comment:
@cheme I don't really have understood this comment. If you know what this should mean, I can add it back.

Reply:
Yeah, it is not clear. It was a note for user who want to check proof: they should not use this method and check proof from the resulting `StorageProof`.
They should check the proof directly from the `MemoryDb` used by `decode_compact` method.
Generally by looking at the api, one could thing unpack first then check, but that would build the memory db twice.